mdpow/analysis/ensemble.py

```diff
@@ -0,0 +1,499 @@
+            elif (file.endswith('gro') or file.endswith('.tpr') or file.endswith('gro.b2z') or file.endswith('gro.gz'))\
+                    and self.top_dict is None:
+                # Saving topology directories
+                self.top.append(file)
+
+        if len(self.top) == 0 or len(self.trj) == 0:
+            logger.warning('No MD files detected in %s', os.curdir)
+            raise NoDataWarning
+
+        self.trj = self._sort_trajectories(self.trj)
+        if len(self.top) > 1:
+            self.top = self._sort_topologies(self.top)
+
+    def keys(self):
+        """Returns list of system keys"""
+        return self._keys
+
+    def _build_ensemble(self):
+        """Goes through given mdpow directory and attempts to build universe in the lambda directories.
+        Run if dirname is passed into __init__. First enters FEP directory, then goes through solvent
+        and interaction directories to search lambda directories for system files."""
+        fep_dir = os.path.join(self._ensemble_dir, 'FEP')
+        for solvent in self._solvents:  # Ugly set of loops, may have to find way to clean up
+            for dirs in self._interactions:  # Attribute folder names
+                int_dir = os.path.join(fep_dir, solvent, dirs)
+                if os.path.exists(int_dir):
+                    with in_dir(int_dir, create=False):  # Entering attribute folders
+                        logger.info("Searching %s directory for systems", os.curdir)
+                        files = os.listdir(os.curdir)
+                        for file in files:  # Traversing lambda directories
+                            if os.path.isdir(file):
+                                with in_dir(file, create=False):
+                                    try:
+                                        self._load_universe_from_dir(solvent)
+                                        self.add_system((solvent, dirs, file), topology=self.top[0],
+                                                        trajectory=self.trj)
+                                    except NoDataWarning:
+                                        logger.warning('Failed to load universe in %s', file)
+                                        continue
+                else:
+                    logger.error('%r directory does not exist', int_dir)
+                    raise NoDataError
+
+    def add_system(self, key, topology=None, trajectory=None, universe=None):
+        """Adds system from universe object for trajectory and topology files
+
+        Takes specified key and either existing mda.Universe object or
+        trajectory and topology path."""
+        if not universe is None:
+            self._ensemble[key] = universe
+            self._keys.append(key)
+            self._num_systems += 1
+            return
+        if isinstance(trajectory, list):
+            # setting abs paths for trajectory
+            trajectory = [os.path.abspath(trj) for trj in trajectory]
```

Comment:
move into `load_universe_from_dirs()`

Reply:
done, I'll add a note in the docs as well so users know to do this too.